syncer/syncer.go

```diff
@@ -1856,6 +1856,8 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 			case *ast.TruncateTableStmt:
 				ec.tctx.L().Info("ignore truncate table statement in shard group", zap.String("event", "query"), zap.String("statement", sqlDDL))
 				continue
+			case *ast.RenameTableStmt:
```

Comment:
we heavily rely on "an optimistic sharding task should sync no other tables than `table-route`", if user both sync sharding tables and non sharding tables in one task, non sharding tables will not permitted to RENAME.

How about we give an error when user both specify block-allow list and table routes, to ensure user add a minimal sharding task that will not sync non-sharding tables? we might talk to PM if you think that's reasonable. @GMHDBJD @lichunzhu 

could join us after your vacation @gozssky

Reply:
(but https://github.com/pingcap/dm/issues/1517 is still not solved)

Reply:
Seems there is not associated between #1517 and rename table? We will support rename in both pessimistic and optimistic later. It's no conflict that we can also support #1517 later.

Reply:
yes, what do you think "this PR also forbid user RENAME a non sharding table in a optimistic sharding task", could we raise an error when user both specify block-allow list and table routes (use table routes as allow list)?

Reply:
We have other stmts not supported for a non sharding table in shard mode(truncate statement). So it's not a problem of Rename.
> raise an error when user both specify block-allow list and table routes 

We can support it in another pr or #1517? Maybe task to PM will be better.